subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -90,9 +90,26 @@ The ability to change the allowed usage of a configuration after creation will b
 ==== Reserved configuration names
 
 Configuration names "detachedConfiguration" and "detachedConfigurationX" (where X is any integer) are reserved for internal use when creating detached configurations.
+Using these names for configurations will now produce a warning.
 
 The ability to create non-detached configurations with these names will be removed in Gradle 9.0.
 
+[[detached_configurations_with_project_dependencies]]
+==== Detached configurations with project dependencies
+
+Adding a dependency on the project which contains it to a detached configuration is deprecated.
+This can result in unintuitive behavior.
```

Comment:
💭  Vague, could we add an example?

Reply:
I have one just below.

Reply:
Reading this again, I think maybe you meant an example of the unintuitive behavior, not the usage.

I'll see if I can add a short example.